drivers/azure/azure.go

```diff
@@ -0,0 +1,464 @@
+package azure
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+
+	"github.com/docker/machine/drivers/azure/azureutil"
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/mcnflag"
+	"github.com/docker/machine/libmachine/state"
+
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+)
+
+const (
+	defaultAzureEnvironment     = "AzurePublicCloud"
+	defaultAzureResourceGroup   = "docker-machine"
+	defaultAzureSize            = "Standard_A2"
+	defaultAzureLocation        = "westus"
+	defaultSSHUser              = "ubuntu" // 'root' not allowed on Azure
+	defaultDockerPort           = 2376
+	defaultAzureImage           = "canonical:UbuntuServer:14.04.3-LTS:latest"
+	defaultAzureVNet            = "docker-machine-vnet"
+	defaultAzureSubnet          = "docker-machine"
+	defaultAzureSubnetPrefix    = "192.168.0.0/16"
+	defaultStorageType          = storage.StandardLRS
+	defaultAzureAvailabilitySet = "docker-machine"
+)
+
+const (
+	flAzureEnvironment     = "azure-environment"
+	flAzureSubscriptionID  = "azure-subscription-id"
+	flAzureResourceGroup   = "azure-resource-group"
+	flAzureSSHUser         = "azure-ssh-user"
+	flAzureDockerPort      = "azure-docker-port"
+	flAzureLocation        = "azure-location"
+	flAzureSize            = "azure-size"
+	flAzureImage           = "azure-image"
+	flAzureVNet            = "azure-vnet"
+	flAzureSubnet          = "azure-subnet"
+	flAzureSubnetPrefix    = "azure-subnet-prefix"
+	flAzureAvailabilitySet = "azure-availability-set"
+	flAzurePorts           = "azure-open-port"
+	flAzurePrivateIPAddr   = "azure-private-ip-address"
+	flAzureUsePrivateIP    = "azure-use-private-ip"
+	flAzureNoPublicIP      = "azure-no-public-ip"
+)
+
+const (
+	driverName = "azure"
+	sshPort    = 22
+)
+
+// Driver represents Azure Docker Machine Driver.
```

Comment:
@ppadala all these Azure concepts can be googled very easily and they are not cryptic (so that only msft employees would understand). I don't see the point of adding a ton of "how Azure works" comments here. godoc might not be the right place for an Azure lecture. :)